lua-examples/ao-standard-token/token.lua

```diff
@@ -137,3 +139,45 @@ handlers.add('mint', handlers.utils.hasMatchingTag('Action', 'Mint'), function(m
     })
   end
 end)
+
+function verifyEccAddress(address)
+    if isEthereumAddress(address) then
+    -- EIP-55
+        local mixedAddress = toChecksumAddress(address)
+        return mixedAddress == address
+    end
+        return true
+end
+
+function isEthereumAddress(address)
```

Comment:
I've been using this:

function isEthereumAddress(address)
	return #address == 42 and string.match(address, "^0x[%x]+$") ~= nil
end

But I should perhaps add the `type(address) == "string"` to have:

function isEthereumAddress(address)
	return type(address) == "string" and #address == 42 and string.match(address, "^0x[%x]+$") ~= nil
end

Reply:
This regular expression is more concise and my implementation is a bit more readable, I'm not very familiar with regulars, so the implementation is a bit more readable I guess